static/recipe.html

```diff
@@ -115,29 +118,53 @@ <h1>Create a new recipe</h1>
     </div>
 
     <script>
+        var allProducts = [];
+        var allIngredients = [];
+
         function loadData() {
-            loadProducts();
+            loadProducts(onProductsLoad);
             loadTableware();
             calculate();
```

Comment:
maybe remove this calculate? we don't really have data just yet - and you call calculate in loadProducts anyways.